subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -456,6 +479,13 @@ int bt_mesh_adv_enable(void)
 		if (err) {
 			return err;
 		}
+
+		if (IS_ENABLED(CONFIG_BT_LL_SOFTDEVICE) && adv->tag & BT_MESH_FRIEND_ADV) {
```

Comment:
I think you need to wrap body of `set_adv_randomness` into ifdef, not the whole function, to use `IS_ENABLED`.